mpl_infer_tiles_GPU_new.py

```diff
@@ -69,146 +78,179 @@ def run(self):
-
-        import pickle
-        worker_root = MPL_Config.WORKER_ROOT
-        db_file_path = os.path.join(worker_root, "neighbors/%s_polydict_%d.pkl" % (self.image_name,self.gpu_id))
-        dbfile = open(db_file_path, 'wb')
+            # get the upper left x y of the image
+            ul_row_divided_img = job_data[0][2]
+            ul_col_divided_img = job_data[0][3]
+            tile_no = job_data[0][4]
+            image = job_data[1]
+
+            results = model.detect([image], verbose=False)
+
+            r = results[0]
+
+            if len(r["class_ids"]):
+                count_p = 0
+
+                for id_masks in range(r["masks"].shape[2]):
+                    # read the mask
+                    mask = r["masks"][:, :, id_masks]
+                    padded_mask = np.zeros(
+                        (mask.shape[0] + 2, mask.shape[1] + 2), dtype=np.uint8
+                    )
+                    padded_mask[1:-1, 1:-1] = mask
+                    class_id = r["class_ids"][id_masks]
+
+                    try:
+                        contours = find_contours(padded_mask, 0.5, "high")[
+                            0
+                        ] * np.array([[self.y_resolution, self.x_resolution]])
+                        contours = contours + np.array(
+                            [[float(ul_row_divided_img), float(ul_col_divided_img)]]
+                        )
+                        # swap two cols
+                        contours.T[[0, 1]] = contours.T[[1, 0]]
+                        # write shp file
+                        w_final.poly([contours.tolist()])
+                        w_final.record(Class=class_id)
+
+                    except:
+                        contours = []
+                        pass
+
+                    count_p += 1
+
+            dict_polygons[int(tile_no)] = [r["masks"].shape[2]]
+
+            if self.config.LOGGING:
+                print(
+                    f"## {count} of {total} ::: {len(r['class_ids'])}  $$$$ {r['class_ids']}"
+                )
+                sys.stdout.flush()
+
+        worker_root = self.config.WORKER_ROOT
+        db_file_path = os.path.join(
+            worker_root,
+            "neighbors/%s_polydict_%d.pkl" % (self.image_name, self.process_counter),
+        )
+        dbfile = open(db_file_path, "wb")
         pickle.dump(dict_polygons, dbfile)
         dbfile.close()
         w_final.close()
-
-def inference_image(POLYGON_DIR,
-                    weights_path,
-                    output_shp_root,
-                    file1,file2,image_name):
-
-    f1 = h5py.File(file1, 'r')
-    f2 = h5py.File(file2, 'r')
-
-    values = f2.get('values')
+        self.input_queue.task_done()
+        print("Exiting Process %d" % self.process_counter)
+
+
+def inference_image(
+    config: MPL_Config,
+    POLYGON_DIR: str,
+    weights_path: str,
+    output_shp_root: str,
+    file1: str,
+    file2: str,
+    image_name: str,
+):
+    f1 = h5py.File(file1, "r")
```

Comment:
what are these two files exactly?

Reply:
My understanding is that these two files are what Amal was referring to in the deep dive meeting when he said they split up the images into two files: one for holding the image data and the other one holds information about how to stitch the images back together to go from tiles to full satelite scene afterwards.